pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -785,3 +784,67 @@ func (a Adapter) ExecCMDInContainer(componentInfo common.ComponentInfo, cmd []st
 func (a Adapter) ExtractProjectToComponent(componentInfo common.ComponentInfo, targetPath string, stdin io.Reader) error {
 	return a.Client.GetKubeClient().ExtractProjectToComponent(componentInfo.ContainerName, componentInfo.PodName, targetPath, stdin)
 }
+
+// Deploy executes the 'deploy' command defined in a devfile
+func (a Adapter) Deploy() error {
+	commands, err := a.Devfile.Data.GetCommands(parsercommon.DevfileOptions{
+		CommandOptions: parsercommon.CommandOptions{
+			CommandGroupKind: devfilev1.DeployCommandGroupKind,
+		},
+	})
+	if err != nil {
+		return nil
+	}
+
+	if len(commands) == 0 {
+		return errors.New("error deploying, no default deploy command found in devfile")
+	}
+
+	if len(commands) > 1 {
+		return errors.New("more than one default deploy command found in devfile, should not happen")
+	}
```

Comment:
Wondering why this shouldn't be in the `Validate` function of `pkg/odo/cli/deploy/deploy.go`.

Reply:
When seeing `deployCmd := commands[0]`, I prefer to check as nearly as possible that the element at this index exists and is valid